sdk/src/client/api/high_level.rs

```diff
@@ -15,40 +15,43 @@ use crate::{
     },
     types::block::{
         address::Bech32Address,
-        core::Block,
         input::{Input, UtxoInput, INPUT_COUNT_MAX},
         output::OutputWithMetadata,
         payload::{transaction::TransactionId, Payload},
         slot::SlotIndex,
-        BlockId,
+        Block, BlockId, BlockWrapper, Error as BlockError,
     },
     utils::unix_timestamp_now,
 };
 
 impl Client {
     /// Get the inputs of a transaction for the given transaction id.
     pub async fn inputs_from_transaction_id(&self, transaction_id: &TransactionId) -> Result<Vec<OutputWithMetadata>> {
-        let block = self.get_included_block(transaction_id).await?;
-
-        let inputs = match block.payload() {
-            Some(Payload::Transaction(t)) => t.essence().inputs(),
-            _ => {
-                unreachable!()
-            }
-        };
-
-        let input_ids = inputs
-            .iter()
-            .map(|i| match i {
-                Input::Utxo(input) => *input.output_id(),
-            })
-            .collect::<Vec<_>>();
-
-        self.get_outputs_with_metadata(&input_ids).await
+        let wrapper = self.get_included_block(transaction_id).await?;
+
+        if let Block::Basic(block) = wrapper.block() {
+            let inputs = match block.payload() {
+                Some(Payload::Transaction(t)) => t.essence().inputs(),
+                _ => {
+                    unreachable!()
+                }
+            };
+
+            let input_ids = inputs
+                .iter()
+                .map(|i| match i {
+                    Input::Utxo(input) => *input.output_id(),
+                })
+                .collect::<Vec<_>>();
+
+            self.get_outputs_with_metadata(&input_ids).await
+        } else {
+            Err(BlockError::InvalidBlockKind(wrapper.block().kind()).into())
```

Comment:
Maybe found/expected?

Reply:
So a different error? What name?

Reply:
Or just update this error?

Reply:
Ah, then yeah a different error probably is warranted

Reply:
Added a new one